management/database/client.go

```diff
@@ -0,0 +1,154 @@
+const (
+	azureCreateDatabaseServerUrl = "services/sqlservers/servers"
+	azureListDatabaseServersUrl  = "/services/sqlservers/servers"
+	azureDeleteDatabaseServerUrl = "/services/sqlservers/servers/%s"
+
+	azureCreateDatabaseUrl = "services/sqlservers/servers/%s/databases"
+	azureGetDatabaseUrl    = "/services/sqlservers/servers/%s/databases/%s"
+	azureListDatabasesUrl  = "/services/sqlservers/servers/%s/databases?contentview=generic"
+	azureDeleteDatabaseUrl = "/services/sqlservers/servers/%s/databases/%s"
+
+	errParamNotSpecified = "Parameter %s is not specified."
+)
+
+// DatabaseClient defines various database CRUD operations.
+// It contains a management.Client for making the actual http calls.
+type DatabaseClient struct {
+	mgmtClient management.Client
+}
+
+// NewClient returns a new DatabaseClient struct with the provided
+// management.Client as the underlying client.
+func NewClient(mgmtClient management.Client) DatabaseClient {
+	return DatabaseClient{mgmtClient}
+}
+
+// CreateDatabaseServer issues a database server creation request to Azure.
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn505699.aspx
+func (c DatabaseClient) CreateDatabaseServer(params DatabaseServerCreateParams) error {
+	if params.Location == "" {
+		return fmt.Errorf(errParamNotSpecified, "Location")
+	}
+	if params.UserName == "" {
+		return fmt.Errorf(errParamNotSpecified, "UserName")
+	}
+	if params.Password == "" {
+		return fmt.Errorf(errParamNotSpecified, "Password")
+	}
+	if params.Version == "" {
+		return fmt.Errorf(errParamNotSpecified, "")
+	}
+
+	req, err := xml.Marshal(params)
+	if err != nil {
+		return err
+	}
+
+	// NOTE: would be nice to be able to read the response here:
+	_, err = c.mgmtClient.SendAzurePostRequest(azureCreateDatabaseServerUrl, req)
+	return err
+}
+
+// ListDatabaseServers issues a request for a list of all database servers off Azure.
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn505702.aspx
+func (c DatabaseClient) ListDatabaseServers() ([]DatabaseServer, error) {
+	var resp ListDatabaseServersResponse
+
+	data, err := c.mgmtClient.SendAzureGetRequest(azureListDatabaseServersUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	err = xml.Unmarshal(data, &resp)
+	return resp.Servers, err
+}
+
+// DeleteDatabaseServer issues an Azure database server deletion request.
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn505695.aspx
+func (c DatabaseClient) DeleteDatabaseServer(name string) error {
+	if name == "" {
+		return fmt.Errorf(errParamNotSpecified, "name")
+	}
+
+	url := fmt.Sprintf(azureDeleteDatabaseServerUrl, name)
+	_, err := c.mgmtClient.SendAzureDeleteRequest(url)
+	return err
+}
+
+// CreateDatabase issues a database creation request to Azure.
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn505701.aspx
+func (c DatabaseClient) CreateDatabase(params Database) error {
+	if params.Name == "" {
+		return fmt.Errorf(errParamNotSpecified, "Name")
+	}
+	if params.DatabaseServer == "" {
```

Comment:
`Server` definitely should not go into `Database` (which should be `DatabaseCreateParameters`). This signature should read as:

    func (c DatabaseClient) CreateDatabase(server string, params DatabaseCreateParameters)